koku/api/provider/provider_builder.py

```diff
@@ -103,7 +103,7 @@ def _create_context(self):
             except Customer.DoesNotExist:
                 customer = IdentityHeaderMiddleware.create_customer(self.account_number, self.org_id, "POST")
             try:
-                user = User.objects.get(username=username)
+                user = User.objects.get(username=username, customer=customer)
```

Comment:
Looking at where this `user` object is used leads me to believe we can just eliminate it from this method. It's only used to remove a provider when we've received the destroy.provider event from the sources backend. We end up in this logic here:
https://github.com/project-koku/koku/blob/5d8b4b4674c0784f9d503a64f8e3f9eec18c68fd/koku/api/provider/provider_manager.py#L366-L368
which takes us here:
https://github.com/project-koku/koku/blob/5d8b4b4674c0784f9d503a64f8e3f9eec18c68fd/koku/api/provider/provider_manager.py#L228-L230

which as you can see, the only thing we actually care about is the customer (not the actual user).

This `_create_context` method already returns the customer, so I think returning the user here is totally redundant and unnecessary. I think we ought to update the `manager.remove` method to rely on the customer from this method or the customer in the request object.